12/umbraco-commerce/reference/storefront-api/README.md

```diff
@@ -0,0 +1,611 @@
+* **Price** - Returns a price for the product based on session information passed through in headers. See the ["Session" concept detailed above](#concepts).
+* **Stock** - Returns the stock level of the given product.
+* **Variants** - See notes on the [variants value converter](#variants-value-converter) below.
+
+### Variants Value Converter
+
+To help with common scenarios when working with variants, the Variants value converter will return a series of data items used when building a relevant UI.
+
+```json
+{
+    attributes: [
+        {
+            alias: "color",
+            name: "Color",
+            values: [
+                {
+                    alias: "red",
+                    name: "Red"
+                },
+                {
+                    alias: "blue",
+                    name: "Blue"
+                }
+            ]
+        },
+        {
+            alias: "size",
+            name: "Size",
+            values: [
+                {
+                    alias: "md",
+                    name: "Medium"
+                },
+                {
+                    alias: "lg",
+                    name: "Large"
+                }
+            ]
+        }
+    ],
+    items: [
+        {
+            attributes: {
+                color: red,
+                size: md
+            },
+            isDefault: true,
+            content: { }
+        },
+        {
+            attributes: {
+                color: blue,
+                size: lg
+            },
+            isDefault: false,
+            content: { }
+        }
+    ],
+    variantContentUrl: "https://{your_domain}/umbraco/delivery/api/v1/content/item/8df5c8bd-b524-4513-805a-c119fc8090e3/variant"
+}
+```
+
+* `attributes` will contain a list of "in use" attributes which means there is at least one variant content item that makes use of that attribute. These should be used to build the attribute options UI.
+* `items` returns a list of variant items. By default this will return the attribute combinations, and whether it is the default combination but it's content value will be empty. The content value can be populated by expanding the variants property through the Delivery API, however it's important to know this could return a lot of data and be intensive. Instead it is prefered to return the none expanded value and use the `variantContentUrl` to fetch individual content items as they are requested. The `items` collection should also be used to check if a combination exists as whilst the root level `attributes` collection contains all in use attributes, it doesn't mean every possible combination of those attributes exists so you can use the `items` collection to validate a combination.
+* `variantContentUrl` as the URL to specialized Delivery API route that can return a single variant items content based on a passed in attribute combination. 
```

Comment:
```suggestion
* `attributes` will contain a list of "in use" attributes which means there is at least one variant content item that makes use of that attribute. These should be used to build the attribute options UI.
* `items` returns a list of variant items. By default, this will return the attribute combinations, and whether it is the default combination but its content value will be empty. The content value can be populated by expanding the variants property through the Delivery API, however, it's important to know this could return a lot of data and be intensive. Instead, it is preferred to return the non-expanded value and use the `variantContentUrl` to fetch individual content items as they are requested. The `items` collection should also be used to check if a combination exists as whilst the root level `attributes` collection contains all in-use attributes, it doesn't mean every possible combination of those attributes exists so you can use the `items` collection to validate a combination.
* `variantContentUrl` as the URL to a specialized Delivery API route that can return a single variant item content based on a passed-in attribute combination. 
```